src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,205 @@
+                + "ON c.id = cd.car_id WHERE cd.driver_id = ? "
+                + "AND c.is_deleted = FALSE";
+        List<Car> allCars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getCarsStatement = connection
+                         .prepareStatement(getCarsByDriverQuery)) {
+            getCarsStatement.setLong(PARAMETER_INDEX_1, driverId);
+            ResultSet resultSet = getCarsStatement.executeQuery();
+            while (resultSet.next()) {
+                allCars.add(getCarFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get cars by id " + driverId, e);
+        }
+        allCars.forEach(c -> c.setDrivers(getDriverById(c.getId())));
+        return allCars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateQuery = "UPDATE cars SET model = ?, manufacturer_id = ?  "
+                + "WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement updateStatement = connection
+                         .prepareStatement(updateQuery)) {
+            updateStatement.setString(PARAMETER_INDEX_1, car.getModel());
+            updateStatement.setLong(PARAMETER_INDEX_2, car.getManufacturer().getId());
+            updateStatement.setLong(PARAMETER_INDEX_3, car.getId());
+            updateStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update car " + car, e);
+        }
+        deleteRelation(car.getId());
+        recordDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteQuery = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement deleteStatement = connection
+                         .prepareStatement(deleteQuery)) {
+            deleteStatement.setLong(PARAMETER_INDEX_1, id);
+            return deleteStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car by id " + id, e);
+        }
+    }
+
+    private void deleteRelation(Long carId) {
+        String deleteQuery = "DELETE FROM cars_drivers WHERE car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement deleteStatement = connection
+                         .prepareStatement(deleteQuery)) {
+            deleteStatement.setLong(PARAMETER_INDEX_1, carId);
+            deleteStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete relation by carId " + carId, e);
+        }
+    }
+
+    private List<Driver> getDriverById(Long carId) {
```

Comment:
```suggestion
    private List<Driver> getDriversById(Long carId) {
```
you are returning many drivers)